frontend/src/hooks/scrollup-hooks.ts

```diff
@@ -1,7 +1,10 @@
 import { useEffect } from "react";
 
-export function useScrollUp(): any {
+export function useScrollUp(oldStep: number, step: number): any {
   useEffect(() => {
-    window.scrollTo(0, 0);
-  });
+    if (oldStep !== step) {
+      window.scrollTo(0, 0);
+      oldStep = step;
```

Comment:
Should we also pass in setOldStep to this custom hook so that the state setter can be called instead assigning a value to the state?